Switch connection only when the current one is failing [run-systemtest]

hmusum opened the pull request:
Subscribers set error on a connection if it fails, so when having
many subscribers we should not switch unless the current connection
is having errors.

Commit:
Switch connection only when the current one is failing

Subscribers set error on a connection if it fails, so when having
many subscribers we should not switch unless the current connection
is having errors

Commit:
Method should not be public

hakonhall commented on config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java:
```diff
@@ -66,19 +66,31 @@ public synchronized JRTConnection getCurrent() {
-                                                    .filter(JRTConnection::isHealthy)
-                                                    .collect(Collectors.toList());
-        JRTConnection newConnection;
+        if ( ! currentConnection.equals(failingConnection)) return currentConnection;
```
Override `equals()` in JRTConnection or use `==` ?

hmusum replied:
Yup, thanks, will do

hakonhall commented on config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java:
```diff
@@ -66,19 +66,31 @@ public synchronized JRTConnection getCurrent() {
 
     @Override
-    public synchronized JRTConnection switchConnection() {
+    public synchronized JRTConnection switchConnection(Connection failingConnection) {
```
An alternative strategy here may be to:
* Return current connection if unequal to failingConnection, or
* return a random connection != failingConnection.

There would then be no need for isHealthy, or the 3 `last` timestamps.

hmusum replied:
Good suggestion, will implement

Commit:
Simplify code

Add equals and hashcode to JRTConnection
Just switch to a new one if current connection is failing
Removes need for last sucess, error etc. and healthy status

Commit:
Fix mock
